xdsl/dialects/csl.py

```diff
@@ -412,6 +573,25 @@ def print(self, printer: Printer):
         printer.print(" ", self.body)
 
 
+@irdl_op_definition
+class CallOp(IRDLOperation):
+    """
+    Call a regular function or task by name
+    """
+
+    name = "csl.call"
+
+    callee = prop_def(SymbolRefAttr)
+    args = var_operand_def(Attribute)
+    result = opt_result_def(Attribute)
+
+    # TODO(dk949): verify that callee corresponds to a real symbol
+
+    # TODO(dk949): verify that function type of callee matches args and result
+
+    # TODO(dk949): verify that if Call is used outside of a csl.func or csl.task it has a result
```

Comment:
Is this a restriction given by CSL?

Reply:
Yes, a call outside of a function or task is a `comptime` call, so can't have sideeffects. I could've articulated that better. 

It could be done as a pass instead.

Reply:
Hmm, I feel this can be done in the verifier? Not sure either way.

Reply:
We can leave it a todo for now